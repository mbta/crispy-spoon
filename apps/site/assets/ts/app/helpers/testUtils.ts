/* eslint-disable */
import toJson, { Json } from "enzyme-to-json";
import { ReactWrapper } from "enzyme";
import {
  IndexedRouteStop,
  StopTree
} from "../../schedule/components/__schedule";

export const testRouteStopListFromStopTree = (
  tree: StopTree
): IndexedRouteStop[] =>
  Object.values(tree.byId).map((node, index) => ({
    ...node.value,
    routeIndex: index
  }));

export const createReactRoot = (): void => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';
};

export const enzymeToJsonWithoutProps = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false,
    map: info => {
      if (!info.props) {
        return info;
      }

      // @ts-ignore is a <style></style> declaration
      if (info.node.type === "style") {
        return info;
      }
      // @ts-ignore is an element in a <noscript></noscript> declaration
      if (info.node.nodeType == undefined) {
        // @ts-ignore
        return info;
      }
      //@ts-ignore is a React component (not HTML element)
      if (info.node.nodeType.toString().includes("function")) {
        // @ts-ignore
        info.props = undefined;
      }

      return info;
    }
  });

export const toJsonEnzyme = (wrapper: ReactWrapper<any>): Json =>
  toJson(wrapper, {
    noKey: false
  });
