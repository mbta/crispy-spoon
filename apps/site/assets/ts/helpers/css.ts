import {
  isABusRoute,
  isACommuterRailRoute,
  isAGreenLineRoute,
  isASilverLineRoute
} from "../models/route";
import { Route } from "../__v3api";

export const routeToCSSClass = (route: string): string =>
  route.toLowerCase().replace(/[ _]/g, "-");

export const busClass = (route: Route): string =>
  isABusRoute(route) && !isASilverLineRoute(route) ? "bus-route-sign" : "";

export const routeToModeName = (route: Route): string => {
  if (isACommuterRailRoute(route)) return "commuter-rail";
  if (route.type === 4) return "ferry";
  if (route.id === "Red" || route.id === "Mattapan") return "red-line";
  if (route.id === "Orange") return "orange-line";
  if (route.id === "Blue") return "blue-line";
  if (isAGreenLineRoute(route)) return "green-line";
  if (isASilverLineRoute(route)) return "silver-line";
  if (isABusRoute(route)) return "bus";
  return "unknown";
};

export const routeBgClass = (route: Route): string =>
  `u-bg--${routeToModeName(route)}`;
