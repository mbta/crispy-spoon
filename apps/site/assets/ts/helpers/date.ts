// Returns a date in the form of "August 7, 2019"
export const formattedDate = (unformatted: string): string => {
  const [year, month, day] = unformatted
    .split(/-/)
    .map(part => Number.parseInt(part, 10));

  // Remember that months in JS are 0-indexed for some reason, hence "month - 1".
  const parsedDate = new Date(year, month - 1, day);
  return parsedDate.toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC"
  });
};

// Returns a date in the form of "Aug 7"
export const shortDate = (date: Date): string =>
  date.toLocaleDateString("en-US", {
    month: "short",
    day: "numeric",
    timeZone: "UTC"
  });

export default formattedDate;
