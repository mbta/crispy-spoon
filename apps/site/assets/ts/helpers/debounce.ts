// Debounce is meant to be used with event listeners. Use it to delay a call
// until after a burst of events, such as keyboard input.
//
//   usage:
//
//     const callback = (ev: Event) => {}
//     document.addEventListener("keyup", debounce(callback, 250))
//

const debounce = (callback: Function, delay: number): EventListener => {
  let timer: number | undefined;

  return (...args) => {
    window.clearTimeout(timer);
    timer = window.setTimeout(() => callback(...args), delay);
  };
};

export const debouncePromise = (
  callback: Function,
  delay: number
): Function => {
  let timerId: number | undefined;

  return function debounced(...args: unknown[]) {
    if (timerId) {
      window.clearTimeout(timerId);
    }

    return new Promise(resolve => {
      timerId = window.setTimeout(() => resolve(callback(...args)), delay);
    });
  };
};

export default debounce;
