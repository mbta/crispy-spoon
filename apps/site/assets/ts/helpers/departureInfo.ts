import { concat, difference, keyBy, keys, map, sortBy } from "lodash";
import { PredictionWithTimestamp } from "../models/perdictions";
import { ScheduleWithTimestamp } from "../models/schedules";
import { isCancelled, isDelayed } from "./prediction-helpers";
import { DepartureInfo } from "../models/departureInfo";

const departureInfoToTime = (departureInfo: DepartureInfo): Date => {
  // If there isn't a prediction there should be a schedule
  return departureInfo.prediction
    ? departureInfo.prediction.time
    : departureInfo.schedule!.time;
};

const mergeIntoDepartureInfo = (
  schedules: ScheduleWithTimestamp[],
  predictions: PredictionWithTimestamp[]
): DepartureInfo[] => {
  const schedulesByTripId = keyBy(schedules, sch => sch.trip.id);
  const predictionsByTripId = keyBy(predictions, prd => prd.trip.id);

  const scheduleTripIds = keys(schedulesByTripId);
  const predictionTripIds = keys(predictionsByTripId);

  // All the schedules with and without predictions
  const scheduleDirectionInfo = map(
    schedulesByTripId,
    (schedule, tripId): DepartureInfo => {
      // prediction could be undefined
      const prediction = predictionsByTripId[tripId];
      return {
        prediction: prediction,
        schedule: schedule,
        isCancelled: isCancelled(prediction),
        isDelayed: isDelayed(prediction, schedule)
      };
    }
  );

  const unmatchedPredictionTripIds = difference(
    predictionTripIds,
    scheduleTripIds
  );
  // All the predictions with no schedules
  const predictionDirectionInfo = map(
    unmatchedPredictionTripIds,
    (tripId: string): DepartureInfo => {
      const prediction = predictionsByTripId[tripId];
      return {
        prediction: prediction,
        isCancelled: isCancelled(prediction)
      } as DepartureInfo;
    }
  );

  const departureInfos = concat(scheduleDirectionInfo, predictionDirectionInfo);
  const sortedDepartureInfo = sortBy(departureInfos, di => {
    // prioritize sorting by predictions
    if (di.prediction) {
      return di.prediction.time;
    }
    return di.schedule?.time;
  });

  return sortedDepartureInfo;
};

export { mergeIntoDepartureInfo, departureInfoToTime };
