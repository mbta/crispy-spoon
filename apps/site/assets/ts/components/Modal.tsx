import React, {
  ReactElement,
  useLayoutEffect,
  MouseEvent,
  createRef
} from "react";
import { createPortal } from "react-dom";
import createFocusTrap from "focus-trap";
import { handleReactExitKeyPress } from "../helpers/keyboard-events";
import {
  AriaLabel,
  AriaLabelledBy,
  labelOrDescribedBy
} from "../helpers/aria-props";

interface ContentProps {
  closeModal: Function;
}

type RenderCallback = (args: ContentProps) => JSX.Element;

interface Props {
  children: RenderCallback;
  closeText?: string | ReactElement<HTMLElement>;
  role?: "dialog" | "alertdialog";
  ariaLabel: AriaLabel | AriaLabelledBy;
  focusElementId?: string; // a selector string to the DOM node that will receive focus when the model is first opened
  className?: string;
  openState?: boolean;
  closeModal: Function;
}

interface ModalContentProps {
  content: RenderCallback;
  closeText: string | ReactElement<HTMLElement>;
  role: string;
  ariaLabel: AriaLabel | AriaLabelledBy;
  closeModal: Function;
  focusElementId: string;
  bodyPadding: string;
  scrollBarPadding: number;
  className?: string;
}

const ModalContent = ({
  content,
  closeText,
  closeModal,
  role,
  ariaLabel,
  focusElementId,
  bodyPadding,
  scrollBarPadding,
  className = ""
}: ModalContentProps): ReactElement<HTMLElement> => {
  useLayoutEffect(() => {
    // Activate trap and disable scroll on background body
    const trap = createFocusTrap("#modal-cover", {
      initialFocus: `#${focusElementId}`,
      clickOutsideDeactivates: true
    });
    trap.activate();

    const htmlElement = document.getElementsByTagName("html")[0];
    const bodyWrapper = document.getElementById("body-wrapper");
    if (bodyWrapper) {
      // aria-hidden for mobile devices where we want to emulate "focus" behavior
      bodyWrapper.setAttribute("aria-hidden", "true");
      htmlElement.classList.add("modal-open");
      bodyWrapper.style.paddingRight = `${scrollBarPadding}px`;
    }

    // Deactivate
    return () => {
      if (bodyWrapper) {
        bodyWrapper.setAttribute("aria-hidden", "false");
        htmlElement.classList.remove("modal-open");
        bodyWrapper.style.paddingRight = bodyPadding;
      }
      trap.deactivate();
    };
  });

  const ref = createRef<HTMLElement>();
  const onClickAway = (e: MouseEvent): void => {
    if (e.target === ref.current) closeModal();
  };

  return createPortal(
    // This does have a role
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    <aside
      ref={ref}
      aria-modal
      {...labelOrDescribedBy(ariaLabel)}
      role={role}
      id="modal-cover"
      className="c-modal__cover"
      tabIndex={-1}
      onKeyDown={e => handleReactExitKeyPress(e, closeModal)}
      onClick={e => onClickAway(e)}
    >
      <div className={`c-modal ${className}`}>
        <button
          id="modal-close"
          className="btn btn-secondary c-modal__close"
          onClick={() => closeModal()}
          type="button"
        >
          {closeText}
        </button>
        <div className="c-modal__content">{content({ closeModal })}</div>
      </div>
    </aside>,
    document.body
  );
};

interface ModalWithNoScrollProps extends Props {
  priorPadding: string;
  paddingRight: number;
}

const Modal = ({
  children,
  closeText = "Close",
  role = "dialog",
  ariaLabel,
  focusElementId = "modal-close",
  priorPadding,
  paddingRight,
  className,
  openState = false,
  closeModal
}: ModalWithNoScrollProps): ReactElement<HTMLElement> | null =>
  openState ? (
    <ModalContent
      bodyPadding={priorPadding}
      scrollBarPadding={paddingRight}
      content={children}
      closeText={closeText}
      role={role}
      ariaLabel={ariaLabel}
      focusElementId={focusElementId}
      className={className}
      closeModal={closeModal}
    />
  ) : null;

// Wrap modal to account for disabling scroll bar on background body
const WrappedModal = (props: Props): ReactElement<HTMLElement> | null => {
  if (typeof document !== "undefined") {
    const bodyWrapper = document.getElementById("body-wrapper");
    const priorPadding =
      (bodyWrapper && bodyWrapper.style.paddingRight) || "0px";
    const paddingRight = bodyWrapper
      ? window.innerWidth - bodyWrapper.clientWidth
      : 0;
    return (
      <Modal
        {...props}
        priorPadding={priorPadding}
        paddingRight={paddingRight}
      />
    );
  }
  return null;
};

export default WrappedModal;
