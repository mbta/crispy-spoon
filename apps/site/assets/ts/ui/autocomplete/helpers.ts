import { omit } from "lodash";
import { OnSubmitParams } from "@algolia/autocomplete-core";
import { OnStateChangeProps } from "@algolia/autocomplete-js";
import {
  ContentItem,
  Item,
  LocationItem,
  PopularItem,
  RouteItem,
  StopItem
} from "./__autocomplete";
import { isLGDown } from "../../helpers/media-breakpoints";

export function isStopItem(x: Item): x is StopItem {
  return Object.keys(x).includes("stop");
}

export function isRouteItem(x: Item): x is RouteItem {
  return Object.keys(x).includes("route");
}

export function isContentItem(x: Item): x is ContentItem {
  return Object.keys(x).includes("_content_type");
}

export const getTitleAttribute = (item: Item): string[] => {
  if (isStopItem(item)) {
    return ["stop", "name"];
  }
  if (isRouteItem(item)) {
    return ["route", "name"];
  }

  // @ts-ignore
  // eslint-disable-next-line no-underscore-dangle
  return Object.keys(item._highlightResult);
};

const navStateChange: (props: OnStateChangeProps<Item>) => void = ({
  state
}) => {
  // grey out the page and disable scrolling when search is open
  if (isLGDown()) {
    if (state.isOpen) {
      document.documentElement.dataset.navOpen = "true";
    } else {
      delete document.documentElement.dataset.navOpen;
    }
  }
};

export const STATE_CHANGE_HANDLERS: Record<
  string,
  (props: OnStateChangeProps<Item>) => void
> = {
  nav: navStateChange
};

export const SUBMIT_HANDLERS: Record<
  string,
  (props: OnSubmitParams<Item>) => void
> = {
  to_search_page({ state }) {
    window.Turbolinks.visit(`/search?query=${state.query}`);
  }
};

export type WithUrls<T> = T & { urls: Record<string, string> };

// The backend returns all possible URLs, use urlType to get the desired one
export const itemWithUrl = (
  initialItem: WithUrls<LocationItem | PopularItem>,
  urlType: string
): Omit<typeof initialItem, "urls"> => {
  const item = omit(initialItem, "urls");
  item.url = initialItem.urls[urlType];
  return item;
};
