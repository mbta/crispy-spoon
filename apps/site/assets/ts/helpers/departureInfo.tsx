import React from "react";
import {
  chain,
  concat,
  difference,
  find,
  keyBy,
  keys,
  map,
  sortBy
} from "lodash";
import { PredictionWithTimestamp } from "../models/predictions";
import { ScheduleWithTimestamp } from "../models/schedules";
import { isCancelled, isDelayed, isSkipped } from "./prediction-helpers";
import { DepartureInfo } from "../models/departureInfo";
import {
  isABusRoute,
  isACommuterRailRoute,
  isSubwayRoute
} from "../models/route";
import { DirectionId, Route } from "../__v3api";
import DisplayTime from "../stop/components/DisplayTime";
import { getInfoKey } from "../stop/models/displayTimeConfig";
import { RoutePatternWithPolyline } from "../models/route-patterns";

export const SUBWAY = "subway";
export const BUS = "bus";
export const COMMUTER_RAIL = "commuter_rail";
export const FERRY = "ferry";

const toRouteMode = (
  route: Route
): typeof SUBWAY | typeof BUS | typeof COMMUTER_RAIL | typeof FERRY => {
  if (isSubwayRoute(route)) {
    return SUBWAY;
  }
  if (isABusRoute(route)) {
    return BUS;
  }
  if (isACommuterRailRoute(route)) {
    return COMMUTER_RAIL;
  }
  return FERRY;
};

const departureInfoToTime = (departureInfo: DepartureInfo): Date => {
  // If there isn't a prediction there should be a schedule
  return departureInfo.prediction && departureInfo.prediction.time
    ? departureInfo.prediction.time
    : departureInfo.schedule!.time;
};

const getNextUnCancelledDeparture = (
  departureInfos: DepartureInfo[]
): DepartureInfo | undefined => {
  return find(departureInfos, (di: DepartureInfo) => !di.isCancelled);
};

const displayInfoContainsPrediction = (
  departureInfo: DepartureInfo
): boolean => {
  return !!departureInfo.prediction;
};

const mergeIntoDepartureInfo = (
  schedules: ScheduleWithTimestamp[],
  predictions: PredictionWithTimestamp[]
): DepartureInfo[] => {
  const schedulesByTripId = keyBy(schedules, "trip.id");
  const predictionsByTripId = keyBy(predictions, "trip.id");
  const scheduleTripIds = keys(schedulesByTripId);
  const predictionTripIds = keys(predictionsByTripId);

  // All the schedules with and without predictions
  const scheduleDirectionInfo = map(
    schedulesByTripId,
    (schedule, tripId): DepartureInfo => {
      // prediction could be undefined
      const prediction = predictionsByTripId[tripId];
      return {
        prediction,
        schedule,
        route: schedule.route,
        trip: schedule.trip,
        isCancelled: isCancelled(prediction),
        isSkipped: isSkipped(prediction),
        isDelayed: isDelayed(prediction, schedule),
        routeMode: toRouteMode(schedule.route)
      };
    }
  );

  const unmatchedPredictionTripIds = difference(
    predictionTripIds,
    scheduleTripIds
  );
  // All the predictions with no schedules
  const predictionDirectionInfo = map(
    unmatchedPredictionTripIds,
    (tripId: string): DepartureInfo => {
      const prediction = predictionsByTripId[tripId];
      return {
        prediction,
        route: prediction.route,
        trip: prediction.trip,
        isCancelled: isCancelled(prediction),
        isSkipped: isSkipped(prediction),
        routeMode: toRouteMode(prediction.route)
      } as DepartureInfo;
    }
  );

  const departureInfos = concat(scheduleDirectionInfo, predictionDirectionInfo);
  const sortedDepartureInfo = sortBy(departureInfos, di => {
    // prioritize sorting by predictions
    if (di.prediction && di.prediction.time) {
      return di.prediction.time;
    }
    return di.schedule?.time;
  });

  return sortedDepartureInfo;
};

const isAtDestination = (
  stopName: string,
  route: Route,
  directionId: DirectionId
): boolean => {
  const destination = route.direction_destinations[directionId];
  if (destination) {
    return destination.includes(stopName) && !destination.includes(" or ");
  }
  return false;
};

const isCommuterRail = (departureInfo: DepartureInfo): boolean => {
  return departureInfo?.routeMode === "commuter_rail";
};

const DefaultWrapper: React.FunctionComponent<unknown> = ({ children }) => (
  <li>{children}</li>
);
/**
 * From a list of `DepartureInfo[]`, generate a list of `<DisplayTime />`.
 * Each `<DisplayTime />` will recieve the `isCR` and `targetDate` props.
 *
 * This function can optionally
 * - wrap each `<DisplayTime />` in a specified wrapper element, `WrapperEl`
 * - truncate the list to a desired `listLength` value
 */
const departuresListFromInfos = (
  departureInfos: DepartureInfo[],
  isCR: boolean,
  targetDate?: Date,
  listLength?: number,
  omitCancelledAndSkipped = false,
  WrapperEl: typeof DefaultWrapper = DefaultWrapper
): React.ReactElement[] => {
  // optional cutoff time, before which we won't show schedules.
  // just used with subway for now.
  const predictionTimeCutoff = chain(departureInfos)
    .filter(d => d.routeMode === SUBWAY)
    .maxBy("prediction.time")
    .value()?.prediction!.time;

  return chain(departureInfos)
    .reject(
      departure =>
        omitCancelledAndSkipped &&
        (!!departure.isCancelled || !!departure.isSkipped)
    )
    .omitBy(
      ({ prediction, schedule }) =>
        // omit schedule-only departures that are before latest prediction time
        predictionTimeCutoff &&
        !prediction &&
        schedule &&
        schedule.time <= predictionTimeCutoff
    )
    .map(d => (
      <WrapperEl key={getInfoKey(d)}>
        <DisplayTime departure={d} isCR={isCR} targetDate={targetDate} />
      </WrapperEl>
    ))
    .slice(0, listLength)
    .value();
};

const departureInfoInRoutePatterns = (
  departureInfo: DepartureInfo,
  routePatterns: RoutePatternWithPolyline[]
): boolean => {
  const isMatch = !!routePatterns.find(
    rp => rp.id === departureInfo.trip.route_pattern_id
  );
  // This second case is needed for the edge case where the route pattern is
  // canonical but the schedules/predictions are not serving canonical route
  // patterns (e.g. due to shuttles or other disruptions)
  const isSimilar = !!routePatterns.find(
    rp =>
      rp.headsign === departureInfo.trip.headsign &&
      rp.route_id === departureInfo.route.id
  );
  return isMatch || isSimilar;
};

export {
  mergeIntoDepartureInfo,
  departureInfoToTime,
  displayInfoContainsPrediction,
  getNextUnCancelledDeparture,
  isAtDestination,
  isCommuterRail,
  departuresListFromInfos,
  departureInfoInRoutePatterns
};
