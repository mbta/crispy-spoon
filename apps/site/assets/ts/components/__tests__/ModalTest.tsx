import React, { useState } from "react";
import { mount } from "enzyme";
import Modal from "../Modal";
import { enzymeToJsonWithoutProps } from "../../app/helpers/testUtils";

const body = '<div id="body-wrapper"><div id="react-root"></div></div>';

const ariaLabel = { label: "Title for modal content" };
const content = (
  <div id="content">
    <p>South Station is accessible. It has the following features:</p>
    <p>This is a test</p>
  </div>
);

interface State {
  modalOpen: boolean;
}

const StatefulModalConsumer = () => {
  const [state, setState] = useState<State>({
    modalOpen: true
  });

  const closeModal = () => {
    setState({ modalOpen: false });
  };

  return state.modalOpen ? (
    <Modal ariaLabel={ariaLabel} closeModal={closeModal}>
      {content}
    </Modal>
  ) : null;
};

test("Modal close function is called when clicked", () => {
  document.body.innerHTML = body;
  const spy = jest.fn();
  const modal = (
    <Modal ariaLabel={{ elementId: "modal-header" }} closeModal={spy}>
      <h2 id="modal-header">A heading</h2>
    </Modal>
  );
  const wrapper = mount(modal);
  wrapper.find("#modal-close").simulate("click");
  expect(spy).toHaveBeenCalled();
});

test("Modal closes function called with Esc key", () => {
  document.body.innerHTML = body;
  const spy = jest.fn();
  const modal = (
    <Modal ariaLabel={ariaLabel} closeModal={spy}>
      {content}
    </Modal>
  );
  const wrapper = mount(modal);
  wrapper.find("#content").simulate("keyDown", { key: "Escape" });
  expect(spy).toHaveBeenCalled();
});

test("Modal closes function called when click is outside of the modal", () => {
  document.body.innerHTML = body;
  const spy = jest.fn();
  const modal = (
    <Modal ariaLabel={ariaLabel} closeModal={spy}>
      {content}
    </Modal>
  );
  const wrapper = mount(modal);
  wrapper.find("#modal-cover").simulate("click");
  expect(spy).toHaveBeenCalled();
});

test("Modal adds padding to body-wrapper to account for disabling the scroll bar", () => {
  document.body.innerHTML = body;
  const modal = <StatefulModalConsumer />;
  const wrapper = mount(modal);
  expect(document.getElementById("body-wrapper")!.style.paddingRight).toBe(
    "1024px"
  );
  // Close modal
  wrapper.find("#modal-close").simulate("click");
  expect(document.getElementById("body-wrapper")!.style.paddingRight).toBe(
    "0px"
  );
});

test("it renders", () => {
  document.body.innerHTML = body;
  const modal = (
    <Modal ariaLabel={ariaLabel} closeModal={() => {}}>
      {content}
    </Modal>
  );
  const wrapper = mount(modal);
  expect(enzymeToJsonWithoutProps(wrapper)).toMatchSnapshot();
});
