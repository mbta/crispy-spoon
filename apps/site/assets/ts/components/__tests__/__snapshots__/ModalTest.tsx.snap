// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<WrappedModal>
  <Modal>
    <ModalContent>
      <Portal
        containerInfo={
          <body
            class="modal-open"
          >
            <div
              aria-hidden="true"
              id="body-wrapper"
              style="padding-right: 1024px;"
            >
              <div
                id="react-root"
              />
            </div>
            <aside
              aria-label="Title for modal content"
              aria-modal="true"
              class="c-modal__cover"
              id="modal-cover"
              role="dialog"
              tabindex="-1"
            >
              <div
                class="c-modal "
              >
                <button
                  class="btn btn-secondary c-modal__close"
                  id="modal-close"
                  type="button"
                >
                  Close
                </button>
                <div
                  class="c-modal__content"
                >
                  <div
                    id="content"
                  >
                    <p>
                      South Station is accessible. It has the following features:
                    </p>
                    <p>
                      This is a test
                    </p>
                  </div>
                </div>
              </div>
            </aside>
          </body>
        }
      >
        <aside
          aria-label="Title for modal content"
          aria-modal={true}
          className="c-modal__cover"
          id="modal-cover"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="dialog"
          tabIndex={-1}
        >
          <div
            className="c-modal "
          >
            <button
              className="btn btn-secondary c-modal__close"
              id="modal-close"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
            <div
              className="c-modal__content"
            >
              <div
                id="content"
              >
                <p>
                  South Station is accessible. It has the following features:
                </p>
                <p>
                  This is a test
                </p>
              </div>
            </div>
          </div>
        </aside>
      </Portal>
    </ModalContent>
  </Modal>
</WrappedModal>
`;
